libdrgn/language.h

```diff
@@ -198,6 +200,42 @@ drgn_element_format_object_flags(enum drgn_format_object_flags flags)
 		(flags & DRGN_FORMAT_OBJECT_ELEMENT_TYPE_NAMES) >> 2);
 }
 
+/** Return the drgn language spec corresponding to the given dwfl language */
+static const struct drgn_language *drgn_dwarf_to_language(int dwfl_lang)
+{
+	switch (dwfl_lang) {
+	case DW_LANG_C:
+	case DW_LANG_C89:
+	case DW_LANG_C99:
+	case DW_LANG_C11:
+		return &drgn_language_c;
+	default:
+		return &drgn_default_language;
+	}
+}
+
+/**
+ * Deduce language which corresponds with the CU of the given dwarf die.
+ *
+ * @param[in] die Dwarf_Die used to retrieve the current language.
+ */
+static const struct drgn_language *drgn_language_from_dwarf(Dwarf_Die *die)
+{
+	Dwarf_Die cudie;
+	if (dwarf_cu_info(die->cu, NULL, NULL, &cudie, NULL, NULL, NULL, NULL) == 0) {
+		// TODO don't fail silently on failure
```

Comment:
Yeah if we can't get the CU info, something's pretty busted, so this should return a `struct drgn_error *` (and return the language via a return parameter).